Leading zeros for time display.

Skyrant opened the pull request:
Time is now shown with leading zeros.

7:9 is now 07:09
10:5 is now 10:05

Skyrant committed:
Leading zeros for time display.

Time is now shown with leading zeros.

7:9 is now 07:09
10:5 is now 10:05

z16 commented on addons/gametime/gametime.lua:
```diff
@@ -272,8 +272,8 @@ function default_settings()
 windower.register_event('time change', function(new, old)
-	gt.hours = (new / 60):floor()
-	gt.minutes = new % 60
+	gt.hours = string.format("%d",(new / 60):floor()):zfill(2)
```
Should change this to the following for efficiency:

``` lua
    gt.hours = '%02d':format((new / 60):floor())
    gt.minutes = '%02d':format(new % 60)
```

This way strings don't have to be reallocated four additional times and it's shorter code. But honestly, you should probably make this toggleable via settings. The game itself does not use leading zeroes, so some people might be confused by this.

Edit: Actually, I now see how this makes sense because of the minutes being truncated otherwise. I'll pull it once you change those two lines.

z16 replied:
Actually, now that I think about it, you should revert this back entirely. This changes the type (and value) of `gt.hours` and `gt.minutes`, which used to be numbers but are now strings. You should change the display, not the data, i.e. the point where `gt.hours` and `gt.minutes` are displayed in the text box should be adjusted by this.

Skyrant replied:
Problem is that the windower api reference is incomplete and there is no description anywhere to be found how texts.new() works or the texts class in general. If i knew how it works i could maybe adjust this but since no one seems to care to maintain the dev wiki i just leave this as is for myself as i can not stand the time without leading zeros.

EDIT: Also can not use format on the output because the variables are NIL before the first time tick so it throws and error. The values are converted to strings anyway.

z16 replied:
Actually, it's not possible with the `texts` library, I wasn't aware that _GameTime_ uses it. I'll add functionality for that in a bit. In the meantime you can change this to what I said above, that should work as an intermediate solution.

Skyrant committed:
Added toggle for padding the time with zeros.

Added the option 'gt zeros [on/off]' to enable and disable padding the time with zeros for nicer formatting on screen. This is also read from the config and requires a change to the settings.xml.

z16 commented:
I rewrote the addon a bit after I added a formatting option to the texts library. By the way, while it's not a great documentation you can check the source code for the texts library. The way `texts.new` works is explained in there and gives an example of how to use it as well.
